lib/react_client.dart

```diff
@@ -512,20 +512,23 @@ final ReactDartInteropStatics2 _dartInteropStatics2 = (() {
         return value;
       });
 
-  void handleComponentWillUpdate(Component2 component, JsMap jsNextProps, JsMap jsNextState) => zone.run(() {
-        component.componentWillUpdate(
-          new JsBackedMap.backedBy(jsNextProps),
-          new JsBackedMap.backedBy(jsNextState),
+  dynamic handleGetSnapshotBeforeUpdate(Component2 component, JsMap jsPrevProps, JsMap jsPrevState) => zone.run(() {
+        final snapshotValue = component.getSnapshotBeforeUpdate(
+          new JsBackedMap.backedBy(jsPrevProps),
+          new JsBackedMap.backedBy(jsPrevState),
         );
 
-        _updatePropsAndStateWithJs(component, jsNextProps, jsNextState);
+        return snapshotValue;
```

Comment:
Hmm since this is passed through the JS layer, it may need to be wrapped in an object to ensure it doesn't get jsified. I'll try this out when testing locally and let you know.

Reply:
Looks like we're good. Could we add an additional test case for passing a Dart List through this, to ensure it doesn't get minified?

You'll also want to use the `same` matcher to ensure it's the same reference:
```dart
arguments: [expectedProps, expectedState, same(snapshot)].
```